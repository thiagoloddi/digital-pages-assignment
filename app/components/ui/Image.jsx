import React, { Component } from 'react';
import injectSheet from 'react-jss';

const style = {
  container: {
    position: 'relative',
    height: '50px',
    width: '50px'    
  },
  img: {
    height: 'inherit',
    width: 'inherit'
  },
  loading: {
    position: 'absolute',
    top: '35%',
    left: '35%',
    height: '30%',
    width: '30%'
  }
}

 class Image extends Component {

  constructor(props) {
    super(props);
    
    this.state = {
      loading: true
    }
  }

  componentDidMount() {
    this.refs.img.onload = () => {
      this.setState({ loading: false });
    }
  }

  componentWillUnmount() {
    this.refs.img.onload = null;
  }

  render() {
    const { classes, src, className = "" } = this.props;
    const { loading } = this.state;
    const hidden = { visibility: 'hidden ' }; 
    return (
      <div className={`${classes.container} ${className}`}>
        <img style={loading ? hidden : null} ref="img" className={classes.img} src={src} />
        <img style={loading ? null : hidden} className={classes.loading} src={"/assets/progress.gif"} />
      </div>
    )
  }
}

export default injectSheet(style)(Image);