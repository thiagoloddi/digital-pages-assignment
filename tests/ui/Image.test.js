import React from 'react';
import renderer from 'react-test-renderer';
import Image from '../../app/components/ui/Image';

const createNodeMock = e => {
  if(e.type == 'img') {
    return {};
  }
}

it('renders correctly loading', () => {
  const tree = renderer
    .create(<Image src="http://lorempixel.com/400/200/" />, { createNodeMock })
    .toJSON();
  expect(tree).toMatchSnapshot();
});