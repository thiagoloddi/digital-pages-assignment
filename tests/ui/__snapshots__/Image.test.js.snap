// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly loading 1`] = `
<div
  className="Image-container-0-1-1 "
>
  <img
    className="Image-img-0-1-2"
    src="http://lorempixel.com/400/200/"
    style={
      Object {
        "visibility": "hidden ",
      }
    }
  />
  <img
    className="Image-loading-0-1-3"
    src="/assets/progress.gif"
    style={null}
  />
</div>
`;
